apisix/core/ctx.lua

```diff
@@ -175,6 +175,20 @@ do
                     end
                 end
 
+            elseif core_str.has_prefix(key, "post_arg_") then
+                -- only match default post form
+                if request.header(nil, "Content-Type") == "application/x-www-form-urlencoded" then
```

Comment:
Let's add a new section in https://github.com/apache/apisix/blob/master/docs/en/latest/router-radixtree.md

Reply:
@spacewander I've added a brief description, check to see if you need to add a separate paragraph `Howto` for descripte.